# **CAREER ACCELERATOR LP1 - PROJECT**

### **Introduction:**

Ideas, creativity, and execution are essential for a start-up to flourish. But are they enough? Investors provide start-ups and other entrepreneurial ventures with the capital---popularly known as "funding"---to think big, grow rich, and leave a lasting impact. In this project, you are going to analyse funding received by start-ups in India from 2018 to 2021. You will find the data for each year of funding in a separate csv file in the dataset provided. In these files you'll find the start-ups' details, the funding amounts received, and the investors' information.


### **Scenario:**
My team Has been tasked with analyzing the Indian Startup Ecosystem. The analysis should provide insight as to the best course of action for the company.

### **Task:**

Our task is to develop a unique story from this dataset by stating and testing a hypothesis, asking questions, perform analysis and share insights with appropriate visualisations.

# **INDIAN STARTUP ECOSYSTEM ANALYSIS 2018 - 2021**

# **1. Business Understanding**

To be able to understand anything, We must first break it apart and examine it's components before we understand how it works as a whole. The task is to perform an analysis of the 'Indian Start-Up Ecosystem', but what exactly do each of these mean? Let's dive into the definitions of each of the elements in the task;

#### **Definitions** ####
##### **Ecosystem:**
In natural sciences, ‘ecosystems’ are generally defined as a system, or a group of interconnected elements, formed by the interaction of a community of organisms with their environment. 

##### **Startup:**
A startup or start-up is a company or project undertaken by an entrepreneur to seek, develop, and validate a scalable business model. Startups are new businesses that intend to grow large beyond the solo founder. At the beginning, startups face high uncertainty and have high rates of failure, but a minority of them do go on to become successful and influential.

##### **India:** 
India is a country that occupies the greater part of South Asia. India is made up of 28 states and eight union territories, and its national capital is New Delhi. It is the seventh-largest country by area and the most populous country as of June 2023.

#### **So What is a Start-Up Ecosystem and why should we care?**

A startup ecosystem is community of people, startups in their various stages and various types of organizations (funders, governments, etc) in a location (physical or virtual), interacting as a system to create and scale new startups. 

Neither biological nor startup ecosystems can be created, designed or built by an outside actor. While this makes the term ‘start-up ecosystem’ hard to grasp, it does underline that start-ups operate in complex and highly dynamic environments. For this reason, it is particularly important to take sufficient time to analyse and understand the ecosystem before designing interventions to partake in it.

Just like biological ecosystems, a startup ecosystem consists of different elements, which can be individuals, groups, organisations and institutions that form a community by interacting with one another, but also environmental determinants that have an influence on how these actors work and interconnect; in startup ecosystems, these can be laws and policies or cultural norms.

![**A Start-Up Ecosystem**](https://upload.wikimedia.org/wikipedia/commons/thumb/3/35/StartupEcosystem.png/300px-StartupEcosystem.png)

#### **Previous Studies / Research**

In nature, for any and all participants to thrive, the ecosystem must be healthy and in balance. For a company this could be the best indicator for whether to invest in an ecosystem or not. Previous studies and researchers have identified 5 key aspects of an ecosystem that can be tracked to measure it's vibrance and and these are:


**1. What is the Density and ecosystem value?**  \
A first step to mapping an ecosystem is to look at its actual size, growth, and value. This can be tracked by the number of new startups founded in a region during a specific period but also the total combined valuation of all these companies over time, and even break them down by funding year to monitor each cohort. Looking at the number of exits, especially the larger ones are also an interesting indicator of startup success.

**2. How does the Funding activity look in the Ecosystem?** \
To assess the health of a startup ecosystem we need to have an eye on the quality, quantity, and ease of access to funding. To evaluate the ease of access to funding, start tracking early-stage funding rounds. Their volume and growth over time will let us know if start-ups are getting the support they need to take their business off the ground. The location of the investors will help you to identify foreign VCs already investing in your Indian startup ecosystem and allow us to build bridges for potential collaboration and partnerships.

**3. Market reach and scaling opportunities** \
The easiest way to gauge the success of your startups is to watch the unicorns (measured in terms of companies valued at over $1 billion) in your ecosystem. Although it may be a metric not relevant in the future (due to the increase in number of unicorns), it remains an interesting indicator of startup ecosystem success.

**4.Knowledge and innovation** \
Innovation and entrepreneurship often flourish alongside world-class knowledge institutes and R&D incentives. These institutions often foster high-impact innovation, collaboration, and success across sectors. You can measure the level of innovation and new technology in your local ecosystem through research and patent activity, and by keeping tabs on the number of spinouts your local knowledge institutions produce. 

**5. Connectedness, Talent, Diversity, and more…** \
A vibrant ecosystem is not simply a collection of isolated elements, the connections between the elements matter just as much as the elements themselves. The metrics for connectedness and access to quality and diverse talent are a little more complex. You could however look out for the number of accelerators & incubators in your region, on job boards to access the type of talent your startups are looking for the most and on investment heatmaps to understand the breadth of various industries or depth of expertise present in your community.

### **Business Objective** 
To find out whether to invest in the Indian start-up ecosystem or not.

#### **Hypothesis**
Null - The Indian Startup Ecosystem is healthy and worth an investment\
Alternative  - The Indian Startup Ecosystem is weak and not worthy of investment

#### **Key Questions**

Using metrics similar to those of previous researchers enables the company to easily compare the Indian case with other global thereby giving the company a broader worldview and the ability to make a more informed decision. 
This is to mean our Key questions will be influenced heavily by the body of previous research.

**1. What is the Total Value of the Indian Startup Ecosystem?**
* How Many startups were founded in the period
* How Much Money has the ecosystem receive in funding 

**2. How has the Ecosystem changed over time?**
* What is the change in performance year on year
* Which region has the best performance

**3. What is the Success rate of Start-ups in the ecosystem?**
* Are there any unicorns from the ecosystem
* How Many Unicorns

**4. Who is already in the Ecosystem?**
* How many companies are already involved in the ecosystem
* What fields are they invested in

**5. Which is the best performing sector in the ecosystem?**
* Sector with highest amount raised
* Sector with most start-ups


#### **Success Criteria**

1. To produce a dashboard that showcases the metrics monitoring the health of the Indian Start-up Ecosystem.
2. To provide an objective metric that can be used to compare with other startup ecosystems.
3. If decision is to invest, to provide guidance on the best path of investment into the Indian Startup Ecosystem.

# **2. Data Understanding**

### **2.1: Data Preparation**

#### **2.1.1: Importations**

In [867]:
# import all necessary libraries
import os
import pandas as pd
import numpy as np
import pyodbc
from dotenv import dotenv_values

#remove pandas display limits
pd.set_option('display.max_columns', None)

#hide warnings
import warnings

warnings.filterwarnings('ignore')


#confrimation all libraries loaded
print("all libraries loaded successfully")

all libraries loaded successfully


### **2.1.2: Database Connection**

In [868]:
#reading data from database
#Load environment variables from .env file into a dictionary variable
environment_variables=dotenv_values('.env')

# Get the values for the credentials you set in the '.env' file
database = environment_variables.get("DB_NAME")
server = environment_variables.get("SERVER_NAME")
username = environment_variables.get("USERNAME")
password = environment_variables.get("PASSWORD")

#Connecting to the database
connection_string = f"DRIVER={{SQL Server}};SERVER={server};DATABASE={database};UID={username};PWD={password}"

# Using the connect method of the pyodbc library.
# This will connect to the server. 
connection=pyodbc.connect(connection_string)

print("connected successfully")

connected successfully


### **Note:** If the connection stops working, try restarting the kernel

#### **2.1.3: Reading the Data**

##### *YEAR: 2018*

In [869]:
# import 2018 data from GitHub
# Available from Azubi Africa Career Accelerator LP1 Repository as csv

df_2018 = pd.read_csv("https://raw.githubusercontent.com/Azubi-Africa/Career_Accelerator_LP1-Data_Analysis/main/startup_funding2018.csv")

df_2018.head()

Company Name                                           Industry  \
0  TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1  Happy Cow Dairy                               Agriculture, Farming   
2       MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3      PayMe India                        Financial Services, FinTech   
4         Eunimart                 E-Commerce Platforms, Retail, SaaS   

  Round/Series       Amount                          Location  \
0         Seed       250000       Bangalore, Karnataka, India   
1         Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2     Series A  ₹65,000,000           Gurgaon, Haryana, India   
3        Angel      2000000       Noida, Uttar Pradesh, India   
4         Seed            —  Hyderabad, Andhra Pradesh, India   

                                       About Company  
0  TheCollegeFever is a hub for fun, fiesta and f...  
1  A startup which aggregates milk from dairy far...  
2          Leading Online Loans Marketplace in India  
3  PayMe India is an innovative FinTech organizat...  
4  Eunimart is a one stop solution for merchants ...

##### *YEAR: 2019*

In [870]:
# import 2019 data from csv
df_2019 = pd.read_csv("startup_funding2019.csv")
df_2019.head()

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   

                                Founders  \
0                     Shantanu Deshpande   
1  Adamas Belva Syah Devara, Iman Usman.   
2                          Jatin Solanki   
3           Srikanth Iyer, Rama Harinath   
4                 Narayana Reddy Punyala   

                                            Investor     Amount($)  \
0                               Sixth Sense Ventures    $6,300,000   
1                                   General Atlantic  $150,000,000   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4           Innovation in Food and Agriculture (IFA)    $6,000,000   

           Stage  
0            NaN  
1       Series C  
2  Fresh funding  
3       Series D  
4            NaN

##### *YEAR: 2020*

In [871]:
#reading the 2020 SQL table into a dataframe

query='''SELECT * 
        FROM dbo.LP1_startup_funding2020'''
        
df_2020=pd.read_sql(query,connection)

df_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  
0                      Angel investors  200000.0      None     None  
1                      GSF Accelerator  100000.0  Pre-seed     None  
2                       Venture Center       NaN  Pre-seed     None  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  
4  Siana Capital Management, Info Edge  340000.0      None     None

##### *YEAR: 2021*

In [872]:
#reading the 2021 SQL table into a dataframe

query='''SELECT * 
        FROM dbo.LP1_startup_funding2021'''
        
df_2021=pd.read_sql(query,connection)

df_2021.head()

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   

                                        Investor        Amount         Stage  
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A  
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None  
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D  
3                         CDC Group, IDG Capital   $51,000,000      Series C  
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed

**Notes:** \
    1. The data for each year is saved in variables named 'df_year'

## **2.2: Exploratory Data Analysis**

The data provided is expected to have the following columns to be used in the analysis:


|  | **COLUMN NAME** | **DESCRIPTION** | **EXPECTED DATATYPE** |
|--|-----------------|-----------------|-----------------------|
|**1**| **Company** | Name of the company/start-up | Object |
|**2**| **Founded** | Year start-up was founded | Datetime[Y] |
|**3**| **Sector** | Sector/ Industry | Category |
|**4**| **What it does** | Description about Company | Object |
|**5**| **Founders** | Founders of the Company | Object |
|**6**| **Investor** | Investors | Category |
|**7**| **Amount** | Raised funds | float64 / int64 |
|**8**| **Stage** | Round of funding reached | Category |
|**9**| **Location** | City/ Region of Startup | Category |

**Key Assumption**
Based on our business understanding and the key questions asked, we have created the expected datatype column to guide our EDA

In [873]:
#checking the 2018 info
df_2018.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 526 entries, 0 to 525
Data columns (total 6 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Company Name   526 non-null    object
 1   Industry       526 non-null    object
 2   Round/Series   526 non-null    object
 3   Amount         526 non-null    object
 4   Location       526 non-null    object
 5   About Company  526 non-null    object
dtypes: object(6)
memory usage: 24.8+ KB


In [874]:
#checking 2019 info
df_2019.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 89 entries, 0 to 88
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company/Brand  89 non-null     object 
 1   Founded        60 non-null     float64
 2   HeadQuarter    70 non-null     object 
 3   Sector         84 non-null     object 
 4   What it does   89 non-null     object 
 5   Founders       86 non-null     object 
 6   Investor       89 non-null     object 
 7   Amount($)      89 non-null     object 
 8   Stage          43 non-null     object 
dtypes: float64(1), object(8)
memory usage: 6.4+ KB


In [875]:
#checking 2020 info
df_2020.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1055 entries, 0 to 1054
Data columns (total 10 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1055 non-null   object 
 1   Founded        842 non-null    float64
 2   HeadQuarter    961 non-null    object 
 3   Sector         1042 non-null   object 
 4   What_it_does   1055 non-null   object 
 5   Founders       1043 non-null   object 
 6   Investor       1017 non-null   object 
 7   Amount         801 non-null    float64
 8   Stage          591 non-null    object 
 9   column10       2 non-null      object 
dtypes: float64(2), object(8)
memory usage: 82.6+ KB


In [876]:
#checking 2021 info
df_2021.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1209 entries, 0 to 1208
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Company_Brand  1209 non-null   object 
 1   Founded        1208 non-null   float64
 2   HeadQuarter    1208 non-null   object 
 3   Sector         1209 non-null   object 
 4   What_it_does   1209 non-null   object 
 5   Founders       1205 non-null   object 
 6   Investor       1147 non-null   object 
 7   Amount         1206 non-null   object 
 8   Stage          781 non-null    object 
dtypes: float64(1), object(8)
memory usage: 85.1+ KB


##### **Notes:** #####
1. The 2018 Dataset has less columns than the expected and all in the object datatype.
2. The 2019 and 2021 datasets have all expected and similar datatypes across columns the Founded column.
3. The 2020 dataset has more columns than expected and two columns in the expected datatypes (Founded and Amount).


##### **Decisions:** #####
We will clean all years' data separately as the columns are not in expected datatypes.\
To identify each data with year collected, we will add a year column to all datasets.

In [877]:
#adding a year column to identify each year's data
df_2018['year'] = 2018
df_2019['year'] = 2019
df_2020['year'] = 2020
df_2021['year'] = 2021

## **Collaboration Tip:**
Clean the data with a focus of creating a dataframe with an info like the Expected table above.

**Reference:** \
Effective Pandas by Matt Harrison - https://www.youtube.com/watch?v=zgbUk90aQ6A&t=4084s

## **2018 CLEANING**

In [878]:
#checking if column names are as expected
(df_2018
.columns)

Index(['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location',
       'About Company', 'year'],
      dtype='object')

**Observations:**
    The column names are not as expected.\
**Decision**
    Check further on items in the columns.

In [879]:
#Checking columns
cols = ['Company Name', 'Industry', 'Round/Series', 'Amount', 'Location','About Company', 'year']

(df_2018
 [cols]
)

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed       250000       Bangalore, Karnataka, India   
1           Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2       Series A  ₹65,000,000           Gurgaon, Haryana, India   
3          Angel      2000000       Noida, Uttar Pradesh, India   
4           Seed            —  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C    225000000       Bangalore, Karnataka, India   
522     Series A            —           Haryana, Haryana, India   
523         Seed         7500        Mumbai, Maharashtra, India   
524         Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525     Series C     35000000        Chennai, Tamil Nadu, India   

                                         About Company  year  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018  
1    A startup which aggregates milk from dairy far...  2018  
2            Leading Online Loans Marketplace in India  2018  
3    PayMe India is an innovative FinTech organizat...  2018  
4    Eunimart is a one stop solution for merchants ...  2018  
..                                                 ...   ...  
521  Udaan is a B2B trade platform, designed specif...  2018  
522            HappyEasyGo is an online travel domain.  2018  
523  Mombay is a unique opportunity for housewives ...  2018  
524  Droni Tech manufacture UAVs and develop softwa...  2018  
525       Welcome to India's most convenient pharmacy!  2018  

[526 rows x 7 columns]

**Observation:**
1. The dataframe has 6 columns as opposed to the Expected 9. There is probably some missing/incomplete data from 2018.
2. The column names are similar/same to those expected.

In [880]:
#checking nulls
(df_2018
.isnull()
.sum())

Company Name     0
Industry         0
Round/Series     0
Amount           0
Location         0
About Company    0
year             0
dtype: int64

**Observation**
There are no nulls. So the DataFrame is missing some expected columns. Probably they weren't recorded.

In [881]:
#checking dtypes
(df_2018
 [cols]
 .dtypes)

Company Name     object
Industry         object
Round/Series     object
Amount           object
Location         object
About Company    object
year              int64
dtype: object

**Observation:**
The datatypes are not as expected for Industry, Amount and Round/Series columns.\
**Decision**
Delve further into the columns 

In [882]:
(df_2018
 [['Industry','Amount','Round/Series']]
 .describe()
 )

Industry Amount Round/Series
count       526    526          526
unique      405    198           21
top           —      —         Seed
freq         30    148          280

#### **Amount Column Cleaning**

In [883]:
#Dealing with the Amount Column
(df_2018
 ['Amount']
 .values
 )

array(['250000', '₹40,000,000', '₹65,000,000', '2000000', '—', '1600000',
       '₹16,000,000', '₹50,000,000', '—', '—', '—', '—', '—', '—', '—',
       '₹100,000,000', '150000', '1100000', '—', '₹500,000', '6000000',
       '650000', '₹35,000,000', '₹64,000,000', '—', '—', '2000000',
       '2000000', '2000000', '—', '₹20,000,000', '1000000', '—', '650000',
       '1000000', '5000000', '4000000', '₹30,000,000', '1000000',
       '2800000', '—', '150000', '₹40,000,000', '1000000', '1700000', '—',
       '1300000', '2000000', '—', '₹5,000,000', '₹20,000,000',
       '₹40,000,000', '₹20,000,000', '1000000', '₹12,500,000',
       '₹15,000,000', '500000', '5000000', '₹104,000,000', '₹45,000,000',
       '13400000', '—', '500000', '—', '₹25,000,000', '—', '₹26,400,000',
       '—', '—', '₹8,000,000', '₹60,000', '4000000', '9000000', '—',
       '100000', '20000', '—', '1000000', '—', '4000000', '250000', '—',
       '120000', '₹34,000,000', '—', '₹342,000,000', '$143,145', '—',
       '₹600

**Observations:**
1. There are multiple currency symbols in the column.\
    Decision: Extract the currency symbols and put in a separate column
2. There are commas in some values.\
    Decision: Remove the commas 
3. There are '-' in some rows\
    Decision: replace them with Nan 

In [884]:
#defining a function to extract the currrency symbols
def extract_currency(df, column_name, curr_char):
    """
    Checks if values in the specified column that start with the given currency character.
    If they do, extracts the currency character and saves it in a new column.

    Args:
        df (pandas.DataFrame): The DataFrame containing the data.
        column_name (str): The name of the column to check.
        curr_char (str): The special character to look for.

    Returns:
        pandas.DataFrame: The modified DataFrame with the new column.
    """
    # Check if values start with the specified special character
    mask = df[column_name].str.startswith(curr_char)

    # Create a new column with the extracted special character
    df['Currency'] = df.loc[mask, column_name].str[0]

    return df

#extracting and saving the currency symbols to a new column
extract_currency(df_2018, 'Amount', curr_char=('$','₹'))

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed       250000       Bangalore, Karnataka, India   
1           Seed  ₹40,000,000        Mumbai, Maharashtra, India   
2       Series A  ₹65,000,000           Gurgaon, Haryana, India   
3          Angel      2000000       Noida, Uttar Pradesh, India   
4           Seed            —  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C    225000000       Bangalore, Karnataka, India   
522     Series A            —           Haryana, Haryana, India   
523         Seed         7500        Mumbai, Maharashtra, India   
524         Seed  ₹35,000,000        Mumbai, Maharashtra, India   
525     Series C     35000000        Chennai, Tamil Nadu, India   

                                         About Company  year Currency  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018      NaN  
1    A startup which aggregates milk from dairy far...  2018        ₹  
2            Leading Online Loans Marketplace in India  2018        ₹  
3    PayMe India is an innovative FinTech organizat...  2018      NaN  
4    Eunimart is a one stop solution for merchants ...  2018      NaN  
..                                                 ...   ...      ...  
521  Udaan is a B2B trade platform, designed specif...  2018      NaN  
522            HappyEasyGo is an online travel domain.  2018      NaN  
523  Mombay is a unique opportunity for housewives ...  2018      NaN  
524  Droni Tech manufacture UAVs and develop softwa...  2018        ₹  
525       Welcome to India's most convenient pharmacy!  2018      NaN  

[526 rows x 8 columns]

In [885]:
#cleaning the amount column and changing to float datatype
df_2018['Amount'] = (df_2018
                     ['Amount']
                     .str
                     .replace(',|\$|₹','', regex= True) #Remove the comma sign and the currency symbols
                     .replace('—', np.NaN, regex=True) #Replace the - with Null
                     .astype(float) #changing to float datatype
                     )
df_2018

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed     250000.0       Bangalore, Karnataka, India   
1           Seed   40000000.0        Mumbai, Maharashtra, India   
2       Series A   65000000.0           Gurgaon, Haryana, India   
3          Angel    2000000.0       Noida, Uttar Pradesh, India   
4           Seed          NaN  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C  225000000.0       Bangalore, Karnataka, India   
522     Series A          NaN           Haryana, Haryana, India   
523         Seed       7500.0        Mumbai, Maharashtra, India   
524         Seed   35000000.0        Mumbai, Maharashtra, India   
525     Series C   35000000.0        Chennai, Tamil Nadu, India   

                                         About Company  year Currency  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018      NaN  
1    A startup which aggregates milk from dairy far...  2018        ₹  
2            Leading Online Loans Marketplace in India  2018        ₹  
3    PayMe India is an innovative FinTech organizat...  2018      NaN  
4    Eunimart is a one stop solution for merchants ...  2018      NaN  
..                                                 ...   ...      ...  
521  Udaan is a B2B trade platform, designed specif...  2018      NaN  
522            HappyEasyGo is an online travel domain.  2018      NaN  
523  Mombay is a unique opportunity for housewives ...  2018      NaN  
524  Droni Tech manufacture UAVs and develop softwa...  2018        ₹  
525       Welcome to India's most convenient pharmacy!  2018      NaN  

[526 rows x 8 columns]

**Observations:**
The Amounts are in differenct currencies.\
**Decision:**
Convert all Ruppee amounts to USD

* **Key Assumption**: We assume that all amounts without a currency symbol are in USD

In [886]:
#converting ₹ to dollars.
def currency_converter(df, target, check, rate = 0.012):
    '''A function to convert the values of received in Rupees to Dollars.
    It checks the currency symbol from one column and them multiplies the target column with the conversion rate
    
    Args:
        df (pandas.DataFrame): The DataFrame containing the data.
        target (str): The name of the column to convert.
        check (str): The name of the column to check currency symbol.
        rate (float): The conversion rate.

    Returns:
        pandas.DataFrame: The modified DataFrame.
    '''
    # Conversion rate of 1 ruppee to dollar in 2018 was = 0.012 
    
    # Check if the currency symbol matches the desired currency
    mask = df[check] == '₹'  # Modify this condition as needed

    # Multiply the amount column by the exchange rate for matching rows
    df.loc[mask, target] *= rate

    return df

#converting all ruppee amounts to usd
currency_converter(df=df_2018, target='Amount', check='Currency')

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed     250000.0       Bangalore, Karnataka, India   
1           Seed     480000.0        Mumbai, Maharashtra, India   
2       Series A     780000.0           Gurgaon, Haryana, India   
3          Angel    2000000.0       Noida, Uttar Pradesh, India   
4           Seed          NaN  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C  225000000.0       Bangalore, Karnataka, India   
522     Series A          NaN           Haryana, Haryana, India   
523         Seed       7500.0        Mumbai, Maharashtra, India   
524         Seed     420000.0        Mumbai, Maharashtra, India   
525     Series C   35000000.0        Chennai, Tamil Nadu, India   

                                         About Company  year Currency  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018      NaN  
1    A startup which aggregates milk from dairy far...  2018        ₹  
2            Leading Online Loans Marketplace in India  2018        ₹  
3    PayMe India is an innovative FinTech organizat...  2018      NaN  
4    Eunimart is a one stop solution for merchants ...  2018      NaN  
..                                                 ...   ...      ...  
521  Udaan is a B2B trade platform, designed specif...  2018      NaN  
522            HappyEasyGo is an online travel domain.  2018      NaN  
523  Mombay is a unique opportunity for housewives ...  2018      NaN  
524  Droni Tech manufacture UAVs and develop softwa...  2018        ₹  
525       Welcome to India's most convenient pharmacy!  2018      NaN  

[526 rows x 8 columns]

In [887]:
#dropping the deprecated Currency column
(df_2018
 .drop(columns='Currency', inplace = True)
 )

df_2018

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

    Round/Series       Amount                          Location  \
0           Seed     250000.0       Bangalore, Karnataka, India   
1           Seed     480000.0        Mumbai, Maharashtra, India   
2       Series A     780000.0           Gurgaon, Haryana, India   
3          Angel    2000000.0       Noida, Uttar Pradesh, India   
4           Seed          NaN  Hyderabad, Andhra Pradesh, India   
..           ...          ...                               ...   
521     Series C  225000000.0       Bangalore, Karnataka, India   
522     Series A          NaN           Haryana, Haryana, India   
523         Seed       7500.0        Mumbai, Maharashtra, India   
524         Seed     420000.0        Mumbai, Maharashtra, India   
525     Series C   35000000.0        Chennai, Tamil Nadu, India   

                                         About Company  year  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018  
1    A startup which aggregates milk from dairy far...  2018  
2            Leading Online Loans Marketplace in India  2018  
3    PayMe India is an innovative FinTech organizat...  2018  
4    Eunimart is a one stop solution for merchants ...  2018  
..                                                 ...   ...  
521  Udaan is a B2B trade platform, designed specif...  2018  
522            HappyEasyGo is an online travel domain.  2018  
523  Mombay is a unique opportunity for housewives ...  2018  
524  Droni Tech manufacture UAVs and develop softwa...  2018  
525       Welcome to India's most convenient pharmacy!  2018  

[526 rows x 7 columns]

#### **Series Column Cleaning**

In [888]:
# checking the unique values in the column
df_2018['Round/Series'].value_counts()

Round/Series
Seed                                                                                                       280
Series A                                                                                                    73
Angel                                                                                                       37
Venture - Series Unknown                                                                                    37
Series B                                                                                                    20
Series C                                                                                                    16
Debt Financing                                                                                              13
Private Equity                                                                                              10
Corporate Round                                                                                    

**Observation:**
There is a google link which could causing the datatype to be different from expected\
**Decision**
clear it

In [889]:
#Cleaning the column
df_2018['Stage'] = (df_2018
                     ['Round/Series']
                     .str
                     .replace(r'http\S+|www\S+', '', regex=True) #cleaning the url and replacing with space
                     .str
                     .strip() # removing empty space
                     .astype('category') #converting to categorical datatype
                     )

df_2018['Stage'].value_counts() #confirming the url has been removed

Stage
Seed                        280
Series A                     73
Venture - Series Unknown     37
Angel                        37
Series B                     20
Series C                     16
Debt Financing               13
Private Equity               10
Corporate Round               8
Pre-Seed                      6
Series E                      5
Grant                         4
Post-IPO Equity               3
Secondary Market              3
Series D                      3
Post-IPO Debt                 2
Undisclosed                   2
Non-equity Assistance         1
Funding Round                 1
Series H                      1
                              1
Name: count, dtype: int64

In [890]:
#dropping the deprecated Round/Series column
(df_2018
 .drop('Round/Series', axis = 1, inplace=True)
 )

df_2018

Company Name                                           Industry  \
0      TheCollegeFever  Brand Marketing, Event Promotion, Marketing, S...   
1      Happy Cow Dairy                               Agriculture, Farming   
2           MyLoanCare   Credit, Financial Services, Lending, Marketplace   
3          PayMe India                        Financial Services, FinTech   
4             Eunimart                 E-Commerce Platforms, Retail, SaaS   
..                 ...                                                ...   
521              Udaan   B2B, Business Development, Internet, Marketplace   
522  Happyeasygo Group                                    Tourism, Travel   
523             Mombay         Food and Beverage, Food Delivery, Internet   
524         Droni Tech                             Information Technology   
525            Netmeds         Biotechnology, Health Care, Pharmaceutical   

          Amount                          Location  \
0       250000.0       Bangalore, Karnataka, India   
1       480000.0        Mumbai, Maharashtra, India   
2       780000.0           Gurgaon, Haryana, India   
3      2000000.0       Noida, Uttar Pradesh, India   
4            NaN  Hyderabad, Andhra Pradesh, India   
..           ...                               ...   
521  225000000.0       Bangalore, Karnataka, India   
522          NaN           Haryana, Haryana, India   
523       7500.0        Mumbai, Maharashtra, India   
524     420000.0        Mumbai, Maharashtra, India   
525   35000000.0        Chennai, Tamil Nadu, India   

                                         About Company  year     Stage  
0    TheCollegeFever is a hub for fun, fiesta and f...  2018      Seed  
1    A startup which aggregates milk from dairy far...  2018      Seed  
2            Leading Online Loans Marketplace in India  2018  Series A  
3    PayMe India is an innovative FinTech organizat...  2018     Angel  
4    Eunimart is a one stop solution for merchants ...  2018      Seed  
..                                                 ...   ...       ...  
521  Udaan is a B2B trade platform, designed specif...  2018  Series C  
522            HappyEasyGo is an online travel domain.  2018  Series A  
523  Mombay is a unique opportunity for housewives ...  2018      Seed  
524  Droni Tech manufacture UAVs and develop softwa...  2018      Seed  
525       Welcome to India's most convenient pharmacy!  2018  Series C  

[526 rows x 7 columns]

In [891]:
#confirming datatypes are as expected
df_2018.dtypes

Company Name       object
Industry           object
Amount            float64
Location           object
About Company      object
year                int64
Stage            category
dtype: object

#### **Industry Column Cleaning**

In [892]:
#checking unique industries
df_2018['Industry'].value_counts()

Industry
—                                              30
Financial Services                             15
Education                                       8
Information Technology                          7
Health Care, Hospital                           5
                                               ..
Education, Gaming, Training                     1
Automotive, Search Engine, Service Industry     1
Wealth Management                               1
B2B, Farming, Marketplace                       1
Biotechnology, Health Care, Pharmaceutical      1
Name: count, Length: 405, dtype: int64

In [893]:
#overview of column
df_2018['Industry']

0      Brand Marketing, Event Promotion, Marketing, S...
1                                   Agriculture, Farming
2       Credit, Financial Services, Lending, Marketplace
3                            Financial Services, FinTech
4                     E-Commerce Platforms, Retail, SaaS
                             ...                        
521     B2B, Business Development, Internet, Marketplace
522                                      Tourism, Travel
523           Food and Beverage, Food Delivery, Internet
524                               Information Technology
525           Biotechnology, Health Care, Pharmaceutical
Name: Industry, Length: 526, dtype: object

**Observations:**
There are multiple items stacked in the column. May make it confusing to know exact industries\
**Decision** 
Split to each individual industry description and maintain only one

**Key Assumption:**
We assume that the first industry description is the primary industry and ignore the rest

In [894]:
#cleaning the industry column
df_2018['Industry'] = (df_2018
                       ['Industry']
                       .str
                       .split(',', expand=True) #split the descriptions separated by comma
                       .pop(0) #pop only the first description
                       .astype('category')
                       )

df_2018['Industry'].value_counts()

Industry
—                                     30
Health Care                           26
Financial Services                    26
E-Commerce                            22
Apps                                  20
                                      ..
Environmental Consulting               1
Enterprise Software                    1
Enterprise Resource Planning (ERP)     1
Embedded Systems                       1
3D Printing                            1
Name: count, Length: 148, dtype: int64

**Note:**
The value_counts of each industry has increased meaning it can be

#### **Location Column Cleaning**

In [895]:
#overview of column
df_2018['Location']

0           Bangalore, Karnataka, India
1            Mumbai, Maharashtra, India
2               Gurgaon, Haryana, India
3           Noida, Uttar Pradesh, India
4      Hyderabad, Andhra Pradesh, India
                     ...               
521         Bangalore, Karnataka, India
522             Haryana, Haryana, India
523          Mumbai, Maharashtra, India
524          Mumbai, Maharashtra, India
525          Chennai, Tamil Nadu, India
Name: Location, Length: 526, dtype: object

* Just like Industry, the column has multiple values separated by comma. In this case it is geographical regions.\
They could come in handy. 

**Decision**
Split into separate columns and assign them to city, region and country columns respectively

In [896]:
#splitting Location column and reassigning to city, region and country
df_2018[['City', 'Region', 'Country']]=(df_2018
                                      ['Location']
                                      .str
                                      .split(',', expand=True) #split columns
                                      .astype('category') #changing to category datatype
                                      )
#drop the deprecated Location column
df_2018.drop(columns='Location', inplace=True)

df_2018

Company Name                Industry       Amount  \
0      TheCollegeFever         Brand Marketing     250000.0   
1      Happy Cow Dairy             Agriculture     480000.0   
2           MyLoanCare                  Credit     780000.0   
3          PayMe India      Financial Services    2000000.0   
4             Eunimart    E-Commerce Platforms          NaN   
..                 ...                     ...          ...   
521              Udaan                     B2B  225000000.0   
522  Happyeasygo Group                 Tourism          NaN   
523             Mombay       Food and Beverage       7500.0   
524         Droni Tech  Information Technology     420000.0   
525            Netmeds           Biotechnology   35000000.0   

                                         About Company  year     Stage  \
0    TheCollegeFever is a hub for fun, fiesta and f...  2018      Seed   
1    A startup which aggregates milk from dairy far...  2018      Seed   
2            Leading Online Loans Marketplace in India  2018  Series A   
3    PayMe India is an innovative FinTech organizat...  2018     Angel   
4    Eunimart is a one stop solution for merchants ...  2018      Seed   
..                                                 ...   ...       ...   
521  Udaan is a B2B trade platform, designed specif...  2018  Series C   
522            HappyEasyGo is an online travel domain.  2018  Series A   
523  Mombay is a unique opportunity for housewives ...  2018      Seed   
524  Droni Tech manufacture UAVs and develop softwa...  2018      Seed   
525       Welcome to India's most convenient pharmacy!  2018  Series C   

          City           Region Country  
0    Bangalore        Karnataka   India  
1       Mumbai      Maharashtra   India  
2      Gurgaon          Haryana   India  
3        Noida    Uttar Pradesh   India  
4    Hyderabad   Andhra Pradesh   India  
..         ...              ...     ...  
521  Bangalore        Karnataka   India  
522    Haryana          Haryana   India  
523     Mumbai      Maharashtra   India  
524     Mumbai      Maharashtra   India  
525    Chennai       Tamil Nadu   India  

[526 rows x 9 columns]

#### **Final Touches**

In [897]:
df_2018 = (df_2018
           .drop_duplicates() #dropping duplicates
           .drop(columns=['Country','Region']) #drop Country and region column
           .rename(columns = {'Company Name': 'company', 'Industry': 'sector', 
                              'About Company': 'what it does', 'City': 'location',
                              'Amount':'amount','Stage':'stage'}) #rename columns
           ) 

df_2018.dtypes

company           object
sector          category
amount           float64
what it does      object
year               int64
stage           category
location        category
dtype: object

## **2019 CLEANING**

In [898]:
#checking if columns are as expected
(df_2019
 .columns
 )

Index(['Company/Brand', 'Founded', 'HeadQuarter', 'Sector', 'What it does',
       'Founders', 'Investor', 'Amount($)', 'Stage', 'year'],
      dtype='object')

In [899]:
#checking nulls
(df_2019
.isnull()
.sum())

Company/Brand     0
Founded          29
HeadQuarter      19
Sector            5
What it does      0
Founders          3
Investor          0
Amount($)         0
Stage            46
year              0
dtype: int64

* There are nulls in 5 columns. dive further into each column

In [900]:
#checking dtypes
(df_2019
.dtypes)

Company/Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What it does      object
Founders          object
Investor          object
Amount($)         object
Stage             object
year               int64
dtype: object

* The founded column is already in the expected datatype. The other columns need to be checked and cleaned.

#### **Amount Column Cleaning**

In [901]:
(df_2019
 ['Amount($)']
 .unique()
 )

array(['$6,300,000', '$150,000,000', '$28,000,000', '$30,000,000',
       '$6,000,000', 'Undisclosed', '$1,000,000', '$20,000,000',
       '$275,000,000', '$22,000,000', '$5,000,000', '$140,500',
       '$540,000,000', '$15,000,000', '$182,700', '$12,000,000',
       '$11,000,000', '$15,500,000', '$1,500,000', '$5,500,000',
       '$2,500,000', '$140,000', '$230,000,000', '$49,400,000',
       '$32,000,000', '$26,000,000', '$150,000', '$400,000', '$2,000,000',
       '$100,000,000', '$8,000,000', '$100,000', '$50,000,000',
       '$120,000,000', '$4,000,000', '$6,800,000', '$36,000,000',
       '$5,700,000', '$25,000,000', '$600,000', '$70,000,000',
       '$60,000,000', '$220,000', '$2,800,000', '$2,100,000',
       '$7,000,000', '$311,000,000', '$4,800,000', '$693,000,000',
       '$33,000,000'], dtype=object)

* The columns have a currency symbol and commas separating the values. There is also 'Undisclosed' text.

**Decision** To clean the currency symbol and comma and replace undisclosed with None

In [902]:
df_2019['Amount'] = (df_2019
                     ['Amount($)']
                     .str
                     .replace(',|\$','', regex= True) #Remove the comma sign and the currency symbols
                     .replace({'Undisclosed' : None}, regex=True) # replacing Undisclosed with None
                     .astype(float) #converting to float
                     )
df_2019

Company/Brand  Founded HeadQuarter                   Sector  \
0    Bombay Shaving      NaN         NaN                Ecommerce   
1         Ruangguru   2014.0      Mumbai                   Edtech   
2          Eduisfun      NaN      Mumbai                   Edtech   
3          HomeLane   2014.0     Chennai          Interior design   
4          Nu Genes   2004.0   Telangana                 AgriTech   
..              ...      ...         ...                      ...   
84     Infra.Market      NaN      Mumbai                Infratech   
85              Oyo   2013.0    Gurugram              Hospitality   
86       GoMechanic   2016.0       Delhi  Automobile & Technology   
87           Spinny   2015.0       Delhi               Automobile   
88  Ess Kay Fincorp      NaN   Rajasthan                  Banking   

                                         What it does  \
0          Provides a range of male grooming products   
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
..                                                ...   
84  It connects client requirements to their suppl...   
85                Provides rooms for comfortable stay   
86  Find automobile repair and maintenance service...   
87                                Online car retailer   
88              Organised Non-Banking Finance Company   

                                             Founders  \
0                                  Shantanu Deshpande   
1               Adamas Belva Syah Devara, Iman Usman.   
2                                       Jatin Solanki   
3                        Srikanth Iyer, Rama Harinath   
4                              Narayana Reddy Punyala   
..                                                ...   
84                    Aaditya Sharda, Souvik Sengupta   
85                                     Ritesh Agarwal   
86  Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   
88                                     Rajendra Setia   

                                             Investor     Amount($)  \
0                                Sixth Sense Ventures    $6,300,000   
1                                    General Atlantic  $150,000,000   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4            Innovation in Food and Agriculture (IFA)    $6,000,000   
..                                                ...           ...   
84  Tiger Global, Nexus Venture Partners, Accel Pa...   $20,000,000   
85  MyPreferred Transformation, Avendus Finance, S...  $693,000,000   
86                                    Sequoia Capital    $5,000,000   
87  Norwest Venture Partners, General Catalyst, Fu...   $50,000,000   
88     TPG, Norwest Venture Partners, Evolvence India   $33,000,000   

            Stage  year       Amount  
0             NaN  2019    6300000.0  
1        Series C  2019  150000000.0  
2   Fresh funding  2019   28000000.0  
3        Series D  2019   30000000.0  
4             NaN  2019    6000000.0  
..            ...   ...          ...  
84       Series A  2019   20000000.0  
85            NaN  2019  693000000.0  
86       Series B  2019    5000000.0  
87            NaN  2019   50000000.0  
88            NaN  2019   33000000.0  

[89 rows x 11 columns]

In [903]:
#dropping deprecated column
(df_2019
 .drop(columns=('Amount($)'), inplace= True)
 )
df_2019

Company/Brand  Founded HeadQuarter                   Sector  \
0    Bombay Shaving      NaN         NaN                Ecommerce   
1         Ruangguru   2014.0      Mumbai                   Edtech   
2          Eduisfun      NaN      Mumbai                   Edtech   
3          HomeLane   2014.0     Chennai          Interior design   
4          Nu Genes   2004.0   Telangana                 AgriTech   
..              ...      ...         ...                      ...   
84     Infra.Market      NaN      Mumbai                Infratech   
85              Oyo   2013.0    Gurugram              Hospitality   
86       GoMechanic   2016.0       Delhi  Automobile & Technology   
87           Spinny   2015.0       Delhi               Automobile   
88  Ess Kay Fincorp      NaN   Rajasthan                  Banking   

                                         What it does  \
0          Provides a range of male grooming products   
1   A learning platform that provides topic-based ...   
2             It aims to make learning fun via games.   
3               Provides interior designing solutions   
4   It is a seed company engaged in production, pr...   
..                                                ...   
84  It connects client requirements to their suppl...   
85                Provides rooms for comfortable stay   
86  Find automobile repair and maintenance service...   
87                                Online car retailer   
88              Organised Non-Banking Finance Company   

                                             Founders  \
0                                  Shantanu Deshpande   
1               Adamas Belva Syah Devara, Iman Usman.   
2                                       Jatin Solanki   
3                        Srikanth Iyer, Rama Harinath   
4                              Narayana Reddy Punyala   
..                                                ...   
84                    Aaditya Sharda, Souvik Sengupta   
85                                     Ritesh Agarwal   
86  Amit Bhasin, Kushal Karwa, Nitin Rana, Rishabh...   
87  Niraj Singh, Ramanshu Mahaur, Ganesh Pawar, Mo...   
88                                     Rajendra Setia   

                                             Investor          Stage  year  \
0                                Sixth Sense Ventures            NaN  2019   
1                                    General Atlantic       Series C  2019   
2      Deepak Parekh, Amitabh Bachchan, Piyush Pandey  Fresh funding  2019   
3   Evolvence India Fund (EIF), Pidilite Group, FJ...       Series D  2019   
4            Innovation in Food and Agriculture (IFA)            NaN  2019   
..                                                ...            ...   ...   
84  Tiger Global, Nexus Venture Partners, Accel Pa...       Series A  2019   
85  MyPreferred Transformation, Avendus Finance, S...            NaN  2019   
86                                    Sequoia Capital       Series B  2019   
87  Norwest Venture Partners, General Catalyst, Fu...            NaN  2019   
88     TPG, Norwest Venture Partners, Evolvence India            NaN  2019   

         Amount  
0     6300000.0  
1   150000000.0  
2    28000000.0  
3    30000000.0  
4     6000000.0  
..          ...  
84   20000000.0  
85  693000000.0  
86    5000000.0  
87   50000000.0  
88   33000000.0  

[89 rows x 10 columns]

#### **Investor Column Cleaning**

In [904]:
(df_2019
 ['Investor']
 .value_counts() #checking unique investors
 #.astype('category')
)


Investor
Undisclosed                                            3
Trifecta Capital                                       2
Sixth Sense Ventures                                   1
Sunil Gavaskar                                         1
Saama Capital, DSG Consumer Partners                   1
                                                      ..
Paytm, Asian VC fund NPTK, Sabre Partners, Neoplux.    1
Canaan Partners                                        1
IVM Intersurer                                         1
Sequoia Capita                                         1
TPG, Norwest Venture Partners, Evolvence India         1
Name: count, Length: 86, dtype: int64

* There are 86 unique investors for 89 records, there is no need to convert to category as it is inefficient in this case.

In [905]:
df_2019.head(10)

Company/Brand  Founded HeadQuarter           Sector  \
0  Bombay Shaving      NaN         NaN        Ecommerce   
1       Ruangguru   2014.0      Mumbai           Edtech   
2        Eduisfun      NaN      Mumbai           Edtech   
3        HomeLane   2014.0     Chennai  Interior design   
4        Nu Genes   2004.0   Telangana         AgriTech   
5        FlytBase      NaN        Pune       Technology   
6           Finly      NaN   Bangalore             SaaS   
7        Kratikal   2013.0       Noida       Technology   
8       Quantiphi      NaN         NaN        AI & Tech   
9        Lenskart   2010.0       Delhi       E-commerce   

                                        What it does  \
0         Provides a range of male grooming products   
1  A learning platform that provides topic-based ...   
2            It aims to make learning fun via games.   
3              Provides interior designing solutions   
4  It is a seed company engaged in production, pr...   
5                        A drone automation platform   
6  It builds software products that makes work si...   
7  It is a product-based cybersecurity solutions ...   
8  It is an AI and big data services company prov...   
9                            It is a eyewear company   

                                          Founders  \
0                               Shantanu Deshpande   
1            Adamas Belva Syah Devara, Iman Usman.   
2                                    Jatin Solanki   
3                     Srikanth Iyer, Rama Harinath   
4                           Narayana Reddy Punyala   
5                                      Nitin Gupta   
6                        Vivek AG, Veekshith C Rai   
7  Pavan Kushwaha, Paratosh Bansal, Dip Jung Thapa   
8                                   Renuka Ramnath   
9     Peyush Bansal, Amit Chaudhary, Sumeet Kapahi   

                                            Investor          Stage  year  \
0                               Sixth Sense Ventures            NaN  2019   
1                                   General Atlantic       Series C  2019   
2     Deepak Parekh, Amitabh Bachchan, Piyush Pandey  Fresh funding  2019   
3  Evolvence India Fund (EIF), Pidilite Group, FJ...       Series D  2019   
4           Innovation in Food and Agriculture (IFA)            NaN  2019   
5                                        Undisclosed            NaN  2019   
6  Social Capital, AngelList India, Gemba Capital...            NaN  2019   
7         Gilda VC, Art Venture, Rajeev Chitrabhanu.   Pre series A  2019   
8               Multiples Alternate Asset Management       Series A  2019   
9                                           SoftBank       Series G  2019   

        Amount  
0    6300000.0  
1  150000000.0  
2   28000000.0  
3   30000000.0  
4    6000000.0  
5          NaN  
6          NaN  
7    1000000.0  
8   20000000.0  
9  275000000.0

#### **Headquarter Column Cleaning**

In [906]:
df_2019['location'] = (df_2019
                       ['HeadQuarter']
                       #.value_counts() #checking unique cities
                       .astype('category') #converting to category
                    )
df_2019.location

0           NaN
1        Mumbai
2        Mumbai
3       Chennai
4     Telangana
        ...    
84       Mumbai
85     Gurugram
86        Delhi
87        Delhi
88    Rajasthan
Name: location, Length: 89, dtype: category
Categories (17, object): ['Ahmedabad', 'Bangalore', 'Chandigarh', 'Chennai', ..., 'Rajasthan', 'Surat', 'Telangana', 'Uttar pradesh']

* converting to category reduces the entries to 17 categories instead of 89 entries

#### **Sector Column Cleaning**

In [907]:
df_2019['sector'] = (df_2019
                     ['Sector']
                     #.value_counts()
                     .astype('category') #converting to category
                    )
df_2019.sector

0                   Ecommerce
1                      Edtech
2                      Edtech
3             Interior design
4                    AgriTech
               ...           
84                  Infratech
85                Hospitality
86    Automobile & Technology
87                 Automobile
88                    Banking
Name: sector, Length: 89, dtype: category
Categories (52, object): ['AI', 'AI & Tech', 'Accomodation', 'AgriTech', ..., 'Transportation', 'Travel', 'Virtual Banking', 'Yoga & wellness']

* converting to category makes the sector column more efficient since it has 52 objects instead of 89

#### **Stage Column Cleaning**

In [908]:
df_2019['stage']= (df_2019
                   ['Stage']
                   #.unique()
                   .fillna('Undisclosed') #fill na with Undisclosed
                   .astype('category') #changing dtype to category
                   .drop(columns='Stage') #drop deprecated column
                   )

df_2019.stage

0       Undisclosed
1          Series C
2     Fresh funding
3          Series D
4       Undisclosed
          ...      
84         Series A
85      Undisclosed
86         Series B
87      Undisclosed
88      Undisclosed
Name: stage, Length: 89, dtype: category
Categories (16, object): ['Fresh funding', 'Post series A', 'Pre series A', 'Pre-series A', ..., 'Series E', 'Series F', 'Series G', 'Undisclosed']

* converting to category reduces the number to 15 categories instead of 89 entries
* The Nulls may represent undisclosed information so we fill with Undisclosed

#### **Founded Column Cleaning**

In [909]:
#Dealing with nulls
df_2019['founded'] = (df_2019
                      ['Founded']
                      .fillna((df_2019.Founded.mode()[0])) #fill missing values with mode
                      .astype('int')
                      )
df_2019.founded

0     2015
1     2014
2     2015
3     2014
4     2004
      ... 
84    2015
85    2013
86    2016
87    2015
88    2015
Name: founded, Length: 89, dtype: int32

#### **Final Touches**

In [910]:
df_2019 = (df_2019
           .drop_duplicates() #dropping duplicates
           .drop(columns=['HeadQuarter','Sector','Stage','Founded']) #drop deprecated columns
           .rename(columns = {'Company/Brand': 'company', 'Investor':'investor', 
                              'What it does': 'what it does','Amount':'amount','Founders':'founders'}) #rename columns
           ) 

df_2019.dtypes

company           object
what it does      object
founders          object
investor          object
year               int64
amount           float64
location        category
sector          category
stage           category
founded            int32
dtype: object

## **2020 CLEANING**

In [911]:
#checking column types
(df_2020
 .head())

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage column10  year  
0                      Angel investors  200000.0      None     None  2020  
1                      GSF Accelerator  100000.0  Pre-seed     None  2020  
2                       Venture Center       NaN  Pre-seed     None  2020  
3  Venture Catalysts, PointOne Capital  400000.0      None     None  2020  
4  Siana Capital Management, Info Edge  340000.0      None     None  2020

In [912]:
df_2020.dtypes

Company_Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What_it_does      object
Founders          object
Investor          object
Amount           float64
Stage             object
column10          object
year               int64
dtype: object

**Notes:**
1. Founded column is not in the correct data type
2. The dataset has all the expected columns. The HeadQuarter column should be Headquarter
3. There is a column (column 10) that is not required for the analysis - to be dropped.

In [913]:
# Drop the column10 column
df_2020 = (df_2020
           .drop(columns="column10")
           )
df_2020.head()

Company_Brand  Founded HeadQuarter              Sector  \
0    Aqgromalin   2019.0     Chennai            AgriTech   
1      Krayonnz   2019.0   Bangalore              EdTech   
2  PadCare Labs   2018.0        Pune  Hygiene management   
3         NCOME   2020.0   New Delhi              Escrow   
4    Gramophone   2016.0      Indore            AgriTech   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   

                              Investor    Amount     Stage  year  
0                      Angel investors  200000.0      None  2020  
1                      GSF Accelerator  100000.0  Pre-seed  2020  
2                       Venture Center       NaN  Pre-seed  2020  
3  Venture Catalysts, PointOne Capital  400000.0      None  2020  
4  Siana Capital Management, Info Edge  340000.0      None  2020

In [914]:
#checking the shapes of the dataframe
(df_2020
 .shape
 )

(1055, 10)

**Notes:**
1. The dataset has 1055 observations and 10 features, the number of features is more than expected.

In [916]:
# Check for missing values
df_2020.isna().sum()


Company_Brand      0
Founded          213
HeadQuarter       94
Sector            13
What_it_does       0
Founders          12
Investor          38
Amount           254
Stage            464
year               0
dtype: int64

**Observation:**
1. The data has A LOT of many missing values.

In [917]:
# Check for duplicates
df_2020.duplicated().sum()

3

**Observation:**
1. There are 3 duplicates in the data, which will be dropped

In [812]:
# Drop duplicates
#df_2020 = (df_2020
#           .drop_duplicates()
#           )

#### **Amount Column Cleaning**

In [920]:
#filling the NaN values with 0
df_2020['Amount'] = (df_2020
                     ['Amount']
                     .fillna(0)
                     )
df_2020.Amount

0         200000.0
1         100000.0
2              0.0
3         400000.0
4         340000.0
           ...    
1050     1500000.0
1051    13200000.0
1052     8000000.0
1053     8043000.0
1054     9000000.0
Name: Amount, Length: 1055, dtype: float64

#### **Note**
Dropping observations is not a solutions to this data cleaning; the problem will be approached in a different manner.

1. The missing values in the `Amount` column has been filled with `0s`
2. The missing values in the `Founded` columns will be filled the mode of the years.
3. All the missing values in the object columns will be filled with `Unknown`, this will ensure the integrity of the data.
4. Columns such as Sector, Investor and stage is having a wrong datatype. We have to convert them to the Categorical Dtype

#### **Founded Column**

In [921]:
# Fill the missing values of the Founded columns with the mode
founded_mode = df_2020['Founded'].mode()[0]

df_2020['founded'] = (df_2020
                      ['Founded']
                      .fillna(founded_mode)
                      .astype('int') #converting to int
                      )

df_2020.founded

0       2019
1       2019
2       2018
3       2020
4       2016
        ... 
1050    2015
1051    2015
1052    2012
1053    2015
1054    2017
Name: founded, Length: 1055, dtype: int32

In [ ]:
#dropping deprecated column
df_2020 = (df_2020
           .drop(columns='Founded')
           )
df_2020.head(10)

Company_Brand HeadQuarter               Sector  \
0    Aqgromalin     Chennai             AgriTech   
1      Krayonnz   Bangalore               EdTech   
2  PadCare Labs        Pune   Hygiene management   
3         NCOME   New Delhi               Escrow   
4    Gramophone      Indore             AgriTech   
5        qZense   Bangalore             AgriTech   
6  MyClassboard   Hyderabad               EdTech   
7         Metvy     Gurgaon  Networking platform   
8        Rupeek   Bangalore              FinTech   
9     Gig India        Pune        Crowdsourcing   

                                        What_it_does  \
0                       Cultivating Ideas for Profit   
1  An academy-guardian-scholar centric ecosystem ...   
2   Converting bio-hazardous waste to harmless waste   
3                       Escrow-as-a-service platform   
4  Gramophone is an AgTech platform enabling acce...   
5  qZense Labs is building the next-generation Io...   
6  MyClassboard is a full-fledged School / Colleg...   
7  AI driven networking platform for individuals ...   
8  Rupeek is an online lending platform that spec...   
9  GigIndia is a marketplace that provides on-dem...   

                                            Founders  \
0                    Prasanna Manogaran, Bharani C L   
1                   Saurabh Dixit, Gurudutt Upadhyay   
2                                    Ajinkya Dhariya   
3                                      Ritesh Tiwari   
4  Ashish Rajan Singh, Harshit Gupta, Nishant Mah...   
5                       Rubal Chib, Dr Srishti Batra   
6                                     Ajay Sakhamuri   
7                                   Shawrya Mehrotra   
8            Amar Prabhu, Ashwin Soni, Sumit Maniyar   
9                       Aditya Shirole, Sahil Sharma   

                                        Investor      Amount         Stage  \
0                                Angel investors    200000.0          None   
1                                GSF Accelerator    100000.0      Pre-seed   
2                                 Venture Center         NaN      Pre-seed   
3            Venture Catalysts, PointOne Capital    400000.0          None   
4            Siana Capital Management, Info Edge    340000.0          None   
5  Venture Catalysts, 9Unicorns Accelerator Fund    600000.0          Seed   
6                                    ICICI Bank.    600000.0  Pre-series A   
7                                     HostelFund         NaN    Pre-series   
8   KB Investment, Bertelsmann India Investments  45000000.0      Series C   
9      Shantanu Deshpande, Subramaniam Ramadorai   1000000.0  Pre-series A   

   year  founded  
0  2020     2019  
1  2020     2019  
2  2020     2018  
3  2020     2020  
4  2020     2016  
5  2020     2019  
6  2020     2008  
7  2020     2018  
8  2020     2015  
9  2020     2017

#### **Final Touches**

In [ ]:
# Converting the dtype of Sector, Stage, and Investor from Object to categorical.
for col in ['Sector', 'Investor', "Stage", "HeadQuarter"]:
    df_2020[col] = df_2020[col].astype('category')
    
df_2020.dtypes

Company_Brand      object
HeadQuarter      category
Sector           category
What_it_does       object
Founders           object
Investor         category
Amount            float64
Stage            category
year                int64
founded             int32
dtype: object

In [ ]:
#renaming all columns to expected column names
df_2020 = (df_2020
           .rename(columns = {'Company_Brand':'company','HeadQuarter':'location',
                              'Sector':'sector','What_it_does':'what it does',
                              'Founders':'founders','Investor':'investor', 'Amount':'amount',
                              'Stage':'stage'}) #renaming columns
           )
df_2020.dtypes

company           object
location        category
sector          category
what it does      object
founders          object
investor        category
amount           float64
stage           category
year               int64
founded            int32
dtype: object

## **2021 CLEANING**

In [598]:
df_2021.head(10)

Company_Brand  Founded HeadQuarter          Sector  \
0  Unbox Robotics   2019.0   Bangalore      AI startup   
1          upGrad   2015.0      Mumbai          EdTech   
2     Lead School   2012.0      Mumbai          EdTech   
3         Bizongo   2015.0      Mumbai  B2B E-commerce   
4        FypMoney   2021.0    Gurugram         FinTech   
5   Urban Company   2014.0   New Delhi   Home services   
6  Comofi Medtech   2018.0   Bangalore      HealthTech   
7     Qube Health   2016.0      Mumbai      HealthTech   
8        Vitra.ai   2020.0   Bangalore    Tech Startup   
9          Taikee   2010.0      Mumbai      E-commerce   

                                        What_it_does  \
0  Unbox Robotics builds on-demand AI-driven ware...   
1     UpGrad is an online higher education platform.   
2  LEAD School offers technology based school tra...   
3  Bizongo is a business-to-business online marke...   
4  FypMoney is Digital NEO Bank for Teenagers, em...   
5  Urban Company (Formerly UrbanClap) is a home a...   
6   Comofi Medtech is a healthcare robotics startup.   
7  India's Most Respected Workplace Healthcare Ma...   
8  Vitra.ai is an AI-based video translation plat...   
9  Taikee is the ISO-certified, B2B e-commerce pl...   

                                            Founders  \
0                        Pramod Ghadge, Shahid Memon   
1  Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...   
2                         Smita Deorah, Sumeet Mehta   
3            Aniket Deb, Ankit Tomar, Sachin Agrawal   
4                                      Kapil Banwari   
5  Abhiraj Singh Bhal, Raghav Chandra, Varun Khaitan   
6                                         Gururaj KB   
7                                        Gagan Kapur   
8                                     Akash Nidhi PS   
9                 Nidhi Ramachandran, Sachin Chhabra   

                                        Investor        Amount         Stage  \
0                    BEENEXT, Entrepreneur First    $1,200,000  Pre-series A   
1       Unilazer Ventures, IIFL Asset Management  $120,000,000          None   
2               GSV Ventures, Westbridge Capital   $30,000,000      Series D   
3                         CDC Group, IDG Capital   $51,000,000      Series C   
4  Liberatha Kallat, Mukesh Yadav, Dinesh Nagpal    $2,000,000          Seed   
5                                     Vy Capital  $188,000,000          None   
6                              CIIE.CO, KIIT-TBI      $200,000          None   
7                      Inflection Point Ventures   Undisclosed  Pre-series A   
8                              Inflexor Ventures   Undisclosed          None   
9                                           None    $1,000,000          None   

   year  
0  2021  
1  2021  
2  2021  
3  2021  
4  2021  
5  2021  
6  2021  
7  2021  
8  2021  
9  2021

In [925]:
#checking datatypes
df_2021.dtypes

Company_Brand     object
Founded          float64
HeadQuarter       object
Sector            object
What_it_does      object
Founders          object
Investor          object
Amount            object
Stage             object
year               int64
dtype: object

**Observations:**
Some columns aren't in the expected datatype.\
**Decision:**
Dive into each column and clean them.

In [926]:
#checking nulls
df_2021.isna().sum()

Company_Brand      0
Founded            1
HeadQuarter        1
Sector             0
What_it_does       0
Founders           4
Investor          62
Amount             3
Stage            428
year               0
dtype: int64

* There are missing values in most columns, we will fill as appropriate

#### **Founded Column Cleaning**

In [929]:
df_2021['founded'] = (df_2021
                      ['Founded']
                      .fillna((df_2021.Founded.mode()[0])) #filling with mode
                      .astype('int') #converting to int
                      )
df_2021.founded

0       2019
1       2015
2       2012
3       2015
4       2021
        ... 
1204    2019
1205    2015
1206    2019
1207    2015
1208    2019
Name: founded, Length: 1209, dtype: int32

#### **HeadQuarter Column cleaning**

In [930]:
df_2021['location'] = (df_2021
                      ['HeadQuarter']
                      .fillna((df_2021.HeadQuarter.mode()[0])) #filling with mode
                      .astype('category') #converting to category
                      )
df_2021.location

0       Bangalore
1          Mumbai
2          Mumbai
3          Mumbai
4        Gurugram
          ...    
1204     Gurugram
1205    New Delhi
1206    Bangalore
1207     Gurugram
1208    Bangalore
Name: location, Length: 1209, dtype: category
Categories (70, object): ['Ahmadabad', 'Ahmedabad', 'Ambernath', 'Andheri', ..., 'Thiruvananthapuram', 'Trivandrum', 'Vadodara', 'West Bengal']

#### **Sector Column Cleaning**

In [934]:
#checking unique entries
df_2021['Sector'].value_counts()

Sector
FinTech                   122
EdTech                    102
Financial Services         61
E-commerce                 37
Food & Beverages           37
                         ... 
Manufacturing startup       1
Cryptocurrency              1
Vehicle repair startup      1
Advisory firm               1
Staffing & Recruiting       1
Name: count, Length: 254, dtype: int64

In [935]:
df_2021['sector'] = (df_2021
                     ['Sector']
                     .astype('category') #converting to category
                    )
df_2021.sector

0                  AI startup
1                      EdTech
2                      EdTech
3              B2B E-commerce
4                     FinTech
                ...          
1204    Staffing & Recruiting
1205         Food & Beverages
1206       Financial Services
1207                   EdTech
1208       Financial Services
Name: sector, Length: 1209, dtype: category
Categories (254, object): ['AI Chatbot', 'AI company', 'AI startup', 'AR startup', ..., 'Water purification', 'Wholesale', 'Wine & Spirits', 'sports']

#### **What_it_does Column Cleaning**

In [936]:
df_2021['what it does'] = (df_2021
                           ['What_it_does'] #renaming the column
                           )

#### **Founders Column Cleaning**

In [940]:
df_2021['Founders']

0                             Pramod Ghadge, Shahid Memon
1       Mayank Kumar, Phalgun Kompalli, Ravijot Chugh,...
2                              Smita Deorah, Sumeet Mehta
3                 Aniket Deb, Ankit Tomar, Sachin Agrawal
4                                           Kapil Banwari
                              ...                        
1204                          Chirag Mittal, Anirudh Syal
1205                                           Bala Sarda
1206                           Arnav Kumar, Vaibhav Singh
1207                                         Ruchir Arora
1208                        Vishal Chopra, Himanshu Gupta
Name: Founders, Length: 1209, dtype: object

In [938]:
df_2021['founders'] = (df_2021
                       ['Founders'] #renaming columns
                       .fillna('Undisclosed') #filling na with undisclosed
                       )
df_2021.founders.isna().sum()

0

#### **Investor Column Cleaning**

In [ ]:
df_2021['investor'] = (df_2021
                       ['Investor']
                       .fillna()
                       )

## **COMBINING ALL CLEANED YEARS**

In [ ]:
#combining all 4 years into one dataframe
dfs = [df_2019,df_2020,df_2021]

comb_df = pd.concat(dfs)
comb_df

Company/Brand  Founded HeadQuarter                 Sector  \
0     Bombay Shaving      NaN         NaN              Ecommerce   
1          Ruangguru   2014.0      Mumbai                 Edtech   
2           Eduisfun      NaN      Mumbai                 Edtech   
3           HomeLane   2014.0     Chennai        Interior design   
4           Nu Genes   2004.0   Telangana               AgriTech   
...              ...      ...         ...                    ...   
1204             NaN   2019.0    Gurugram  Staffing & Recruiting   
1205             NaN   2015.0   New Delhi       Food & Beverages   
1206             NaN   2019.0   Bangalore     Financial Services   
1207             NaN   2015.0    Gurugram                 EdTech   
1208             NaN   2019.0   Bangalore     Financial Services   

                                           What it does  \
0            Provides a range of male grooming products   
1     A learning platform that provides topic-based ...   
2               It aims to make learning fun via games.   
3                 Provides interior designing solutions   
4     It is a seed company engaged in production, pr...   
...                                                 ...   
1204                                                NaN   
1205                                                NaN   
1206                                                NaN   
1207                                                NaN   
1208                                                NaN   

                                   Founders  \
0                        Shantanu Deshpande   
1     Adamas Belva Syah Devara, Iman Usman.   
2                             Jatin Solanki   
3              Srikanth Iyer, Rama Harinath   
4                    Narayana Reddy Punyala   
...                                     ...   
1204            Chirag Mittal, Anirudh Syal   
1205                             Bala Sarda   
1206             Arnav Kumar, Vaibhav Singh   
1207                           Ruchir Arora   
1208          Vishal Chopra, Himanshu Gupta   

                                               Investor     Amount($)  \
0                                  Sixth Sense Ventures    $6,300,000   
1                                      General Atlantic  $150,000,000   
2        Deepak Parekh, Amitabh Bachchan, Piyush Pandey   $28,000,000   
3     Evolvence India Fund (EIF), Pidilite Group, FJ...   $30,000,000   
4              Innovation in Food and Agriculture (IFA)    $6,000,000   
...                                                 ...           ...   
1204                                    Endiya Partners           NaN   
1205                                           IIFL AMC           NaN   
1206                                       Owl Ventures           NaN   
1207                   Winter Capital, ETS, Man Capital           NaN   
1208                     3one4 Capital, Kalaari Capital           NaN   

              Stage  Year Company_Brand  \
0               NaN  2019           NaN   
1          Series C  2019           NaN   
2     Fresh funding  2019           NaN   
3          Series D  2019           NaN   
4               NaN  2019           NaN   
...             ...   ...           ...   
1204   Pre-series A  2021      Gigforce   
1205       Series D  2021        Vahdam   
1206       Series C  2021  Leap Finance   
1207       Series B  2021  CollegeDekho   
1208       Series A  2021        WeRize   

                                           What_it_does     Amount column10  
0                                                   NaN        NaN      NaN  
1                                                   NaN        NaN      NaN  
2                                                   NaN        NaN      NaN  
3                                                   NaN        NaN      NaN  
4                                                   NaN        NaN      NaN  
...                                                 ..